boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig

```diff
@@ -16,6 +16,12 @@ config BOARD
 	default "nrf5340pdk_nrf5340_cpuapp" if BOARD_NRF5340PDK_NRF5340_CPUAPP || BOARD_NRF5340PDK_NRF5340_CPUAPPNS
 	default "nrf5340dk_nrf5340_cpuapp" if BOARD_NRF5340DK_NRF5340_CPUAPP || BOARD_NRF5340DK_NRF5340_CPUAPPNS
 
+# By default, if we build for a Non-Secure version of the board,
+# force building with TF-M as the Secure Execution Environment.
+
+config BUILD_WITH_TFM
```

Comment:
Why do we want this by default exactly? Is there a reason why the current approach is not suitable?

Reply:
@carlescufi The upstream project, currently, does not have a solution for TEE on nRF5340; this patch is offering TFM as the default solution. For upstream users only.

This patch allows us to run full CI on the _NS board in the Zephyr project, using TFM as the SPE. Currently we have no coverage of _NS builds.

I do understand the implications of it in downstream use of the nRF5340 DK board definition, and I am open to suggestions. 

We can revert this `default y` if you think it is going to cause problems. Then building for _NS needs to manually enable TFM - and I don't see why, since there is no alternative..

I would keep, though the SRAM partition modifications, so we won't need to duplicate overlays everywhere.

Reply:
Just an observation.
Users who disable `BUILD_WITH_TFM` will still be restricted to only 192kB RAM in ns image, unless they create an overlay file themselves.

But is that clear to users ?

Reply:
> Users who disable BUILD_WITH_TFM will still be restricted to only 192kB RAM in ns image

@tejlmand thanks! So that's correct now: even without TF-M there will be less RAM available to the NS image.
Implications:
- **upstream**: nothing really. there is no upstream solution for TEE (other than tf-m) so if anyone wants to build for _NS, they will have to have their own SPE solution. If they are capable of doing that, i don't see why they cannot add an overlay with whatever SRAM they want.
- **downstream**: I believe NCS is using PM, so the default SRAM is anyways overwritten. So I don't see implications either. If there are NCS samples without PM, then I guess people will notice less SRAM in the NS side. Then they will need an overlay.

Reply:
Thanks for the explanation.